templates/puppet/provider.erb

```diff
@@ -60,6 +60,13 @@ Puppet::Type.type(:<%= object.out_name -%>).provide(:google) do
   end
 
   def self.prefetch(resources)
+<% if object.deprecated -%>
+    Puppet.warning [
+      "<%= object.out_name -%> will be deprecated in a future release.",
```

Comment:
This message is mostly fine, but I don't think users will know whether they need full URLs or just partial ones, or even just names.  Can you specify?